tcms/static/js/testrun_actions.js

```diff
@@ -683,6 +654,59 @@ jQ(document).ready(function(){
   });
 });
 
+function drawPercentBar() {
+
+  // TODO: better way to do this
+  const testRunId = window.location.pathname.split('/')[2]
+
+  jsonRPC('TestExecution.filter', {'run_id': testRunId}, testExecutions => {
+    
+    let successCount = 0;
+    let failCount = 0;
+    let allCount = testExecutions.length;
+    let statusCount = {}
+    // TODO: better solution
+    executionStatuses.forEach(s => statusCount[s.name] = { count: 0, id: s.id })
+
+    testExecutions.forEach(testExecution => {
+      const executionStatus = executionStatuses.find(s => s.id === testExecution.status_id)
+
+      if (executionStatus.weight > 0) {
+        successCount++
+      } else if (executionStatus.weight < 0) {
+        failCount++
+      }
+
+      statusCount[executionStatus.name].count++
+    })
+  
+    const completeCount = successCount + failCount;
+    const completePercent = completeCount / allCount * 100;
+    const failurePercent = failCount / completeCount * 100;
+    $(".progress-bar > .percent > .complete-percent").text(`${completePercent}%`)
+    $(".progress-bar > .progress-inner").css("width", `${completePercent}%`)
+    $(".progress-bar > .progress-inner > .progress-failed").css("width", `${failurePercent}%`)
+    $(".js-caserun-total").text(allCount)
+
+    $(".count-per-status-list").remove()
+    $(".count-per-status-container").prepend('<span class="count-per-status-list"></span>')
+    for (var status in statusCount) {
```

Comment:
this looks like it can be a separate function. 
that will make the linters happier and is less prone to breakage when this page gets redesigned, which appears to be quite soon.